src/auth/passkey-config.ts

```diff
@@ -0,0 +1,188 @@
+
+/**
+ * Interface representing the properties to update in a passkey config.
+ */
+export interface PasskeyConfigRequest {
+    /**
+     * An array of website or app origins. Only challenges signed
+     * from these origins will be allowed for signing in with passkeys.
+     */
+    expectedOrigins?: string[];
+}
+
+/**
+ * Response received from the server when retrieving, creating, or updating the passkey config.
+ */
+export interface PasskeyConfigServerResponse {
+    name?: string;
+    rpId?: string;
+    expectedOrigins?: string[];
+}
+
+/**
+ * Request for creating or updating the passkey config on the server.
+ */
+export interface PasskeyConfigClientRequest {
+    rpId?: string;
+    expectedOrigins?: string[];
+}
+
+/**
+ * Configuration for signing in users using passkeys.
+ */
+export class PasskeyConfig {
+  /**
+   * The name of the PasskeyConfig resource.
+   */
+  public readonly name?: string;
+  /**
+   * The relying party ID for passkey verifications.
+   * This cannot be changed once created.
+   */
+  public readonly rpId?: string;
+  /**
+   * The allowed website or app origins.
+   * Only challenges signed from these origins will be allowed for signing in with passkeys.
+   */
+  public readonly expectedOrigins?: string[];
+
+  /**
+   * Validates a passkey config request object and throws an error on failure.
+   * @param isCreateRequest - A boolean indicating if it's a create request or not.
+   * @param passkeyConfigRequest - Passkey config to be set.
+   * @param rpId - (optional) Relying party ID if it's a create request.
+   * @throws FirebaseAuthError - If validation fails.
+   * 
+   * @internal
+   */
+  private static validate(isCreateRequest: boolean, passkeyConfigRequest?: PasskeyConfigRequest, rpId?: string): void {
+    // Validation for creating a new PasskeyConfig.
+    if (isCreateRequest && !validator.isNonEmptyString(rpId)) {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INVALID_ARGUMENT,
+        "'rpId' must be a non-empty string.",
+      );
+    }
+    // Validation for updating an existing PasskeyConfig.
+    if (!isCreateRequest && typeof rpId !== 'undefined') {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INVALID_ARGUMENT,
+        "'rpId' cannot be changed once created.",
+      );
+    }
+    if (!validator.isNonNullObject(passkeyConfigRequest)) {
+      throw new FirebaseAuthError(
+        AuthClientErrorCode.INVALID_ARGUMENT,
+        "'passkeyConfigRequest' must not be null.",
+      );
+    }
+    const validKeys = {
+      expectedOrigins: true,
+    };
+    // Check for unsupported top-level attributes.
+    for (const key in passkeyConfigRequest) {
+      if (!(key in validKeys)) {
+        throw new FirebaseAuthError(
+          AuthClientErrorCode.INVALID_ARGUMENT,
+          `'${key}' is not a valid PasskeyConfigRequest parameter.`,
+        );
+      }
+    }
+    if (!validator.isNonEmptyArray(passkeyConfigRequest.expectedOrigins)) {
```

Comment:
for disable case, user should be allowed to pass in empty list.

Reply:
Consulted backend, the disable workaround does not allow empty list.  resolving this.